x/archivedb/db.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package archivedb
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+// ArchiveDb
```

Comment:
Could we add type assertions at the top of this file similar to here: https://github.com/ava-labs/avalanchego/blob/master/database/prefixdb/db.go#L20?

I think we should have `database.KeyValueReader` (for a specific height) and `database.Batch` also for a specific height.

I think supporting just that can be the first PR here and adding support for an iterator over values at a specific height can/should be a follow on PR.

Reply:
> I think we should have database.KeyValueReader (for a specific height) and database.Batch also for a specific height.

We already have a `database.Batch` for a given height. The height is automatic, and the database keeps track of the current height. That was @patrick-ogrady's suggestion and I like that. There is no way to tamper with the database.